bootkube/resources/charts/calico.yaml

```diff
@@ -9,3 +9,27 @@ calico:
   ipipReadiness: "${ipip_readiness}"
   podCIDR: ${pod_cidr}
   networkEncapsulation: "${network_encapsulation}"
+  hostProtection:
+    enabled: ${host_protection}
+    %{~ if node_cidr != "" ~}
+    nodeCIDR: ${node_cidr}
+    %{~ endif ~}
+    %{~ if length(management_cidrs) > 0 ~}
+    managementCIDRs:
+    %{~ for cidr in management_cidrs ~}
+    - ${cidr}
+    %{~ endfor ~}
+    %{~ endif ~}
+  %{~ if length(host_endpoints) > 0 ~}
+  hostEndpoints:
+  %{~ for h in host_endpoints ~}
+  - name: ${h.name}
+    interfaceName: ${h.interface_name}
+    %{~ if length(h.labels) > 0 ~}
+    labels:
+    %{~ for k, v in h.labels ~}
+      ${k}: ${v}
+    %{~ endfor ~}
+    %{~ endif ~}
+  %{~ endfor ~}
+  %{~ endif ~}
```

Comment:
Is so horrible and error prone this... :-( I understand we can't do anything about it, but I really dislike it :-)

Reply:
@invidian will mark this as solved. Just to make a point :joy:

Reply:
Haha, yeah. I can look if we can use https://www.terraform.io/docs/configuration/functions/yamlencode.html instead of templates.

Reply:
Oh, but this function is currently experimental, so I wouldn't actually use it.